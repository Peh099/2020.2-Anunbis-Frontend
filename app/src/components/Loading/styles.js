import styled, { keyframes } from 'styled-components';

const translateY = keyframes`
    0% {
        transform: translateY(0%);
    }
    50% {
        transform: translateY(-150%);
    }
`;

export const Ball = styled.span`
  width: 10px;
  height: 10px;
  background-color: var(--darkBlue);
  border-radius: 50%;
  animation: ${translateY} 1s ease-in-out infinite;
  animation-delay: ${(props) => props.delay};
`;

export const LoadingDiv = styled.div`
  width: 50px;
  height: fit-content;
  display: grid;
  grid-template-columns: auto auto auto;
`;

export const LoadingBox = styled.div`
  width: 100%;
  height: 20vh;
  background: ${(props) => (props.backColor ? props.backColor : '')};
  display: flex;
  justify-content: center;
  align-items: center;
`;
