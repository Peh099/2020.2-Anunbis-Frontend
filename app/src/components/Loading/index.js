import React from 'react';
import { LoadingDiv, Ball, LoadingBox } from './styles';

export default function Loading() {
  return (
    <LoadingBox data-testid="load-1">
      <LoadingDiv>
        <Ball delay="0" />
        <Ball delay="0.1s" />
        <Ball delay="0.2s" />
      </LoadingDiv>
    </LoadingBox>
  );
}
