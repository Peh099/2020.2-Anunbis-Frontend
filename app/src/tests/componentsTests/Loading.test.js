import renderer from 'react-test-renderer';
import Loading from '../../components/Loading';

describe('Snapshot Loading component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<Loading />).toJSON()
    
    expect(tree).toMatchSnapshot()
    })
})