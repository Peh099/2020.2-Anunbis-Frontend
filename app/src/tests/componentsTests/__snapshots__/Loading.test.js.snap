// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Loading component matches the snapshot 1`] = `
<div
  className="sc-gtsrHT hgaNOb"
>
  <span
    className="sc-bdnxRM gGcvmz"
  />
  <span
    className="sc-bdnxRM hAXQfX"
  />
  <span
    className="sc-bdnxRM hlSWso"
  />
</div>
`;
